models/articles.model.js

```diff
@@ -33,21 +34,42 @@ exports.changeVotesByArticleId = (article_id, inc_votes) => {
     });
 };
 
-exports.findArticles = () => {
-  return db
-    .query(
-      `SELECT articles.article_id, articles.title, articles.author, articles.topic, articles.created_at, articles.votes, COUNT(comments.comment_id)::INT AS comment_count FROM articles
-     JOIN comments 
-     ON comments.article_id = articles.article_id
-     GROUP BY articles.article_id
-     ORDER BY articles.created_at DESC;`
-    )
-    .then((result) => {
-      if (result.rows.length === 0) {
-        return Promise.reject({ status: 404, msg: "not found!" });
-      }
-      return result.rows;
-    });
+exports.findArticles = (sort_by = "created_at", order = "DESC", topic) => {
+  const validSortBy = ["created_at", "article_id", "votes", "title"];
+  const validOrder = ["asc", "desc", "ASC", "DESC"];
+  const validTopics = ["mitch", "cats", "paper"];
+
+  if (!validSortBy.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "bad request!" });
+  }
+
+  if (!validOrder.includes(order)) {
+    return Promise.reject({ status: 400, msg: "bad request!" });
+  }
+
```

Comment:
Previous two if blocks could be combined into one to make the code a bit drier, since they are both returning the same Promise.reject anyway 😄